homeassistant/components/hassio/__init__.py

```diff
@@ -222,6 +222,18 @@ async def async_start_addon(hass: HomeAssistant, slug: str) -> dict:
     return await hassio.send_command(command, timeout=60)
 
 
+@bind_hass
+@api_data
+async def async_restart_addon(hass: HomeAssistant, slug: str) -> dict:
+    """Restart add-on.
+
+    The caller of the function should handle HassioAPIError.
+    """
+    hassio = hass.data[DOMAIN]
+    command = f"/addons/{slug}/restart"
+    return await hassio.send_command(command, timeout=60)
```

Comment:
Some add-ons can take a long time both to shutdown and startup again, especially on slower devices.
Maybe the z-wavejs addon where this is planed now does not have that problem, but I think this should have no timeout.
The frontend sends None for this call for that reason

Reply:
```suggestion
    return await hassio.send_command(command, timeout=None)
```